### PR TITLE
fix(widgetStore): tolerate null/undefined custom widgets from extensions

### PR DESCRIPTION
**Pull request by mattmillerai**
## ELI-5

Some custom nodes have a `getCustomWidgets()` function that's *supposed* to hand
us a list of widgets. A few of them hand us back nothing (null/undefined)
instead. We were trying to read that "nothing" like a list, which crashes with
*"Cannot convert undefined or null to object"* — and because it happens while
the app is still starting up, it can break the whole page. This PR just says
"if there's nothing to register, skip it."

## What

`registerCustomWidgets` called `Object.entries(newWidgets)` directly. When an
extension's `getCustomWidgets()` resolves to `null`/`undefined` (it's typed
non-null, but extensions are untrusted and routinely violate the type), this
throws `TypeError: Cannot convert undefined or null to object`.

The call site in `extensionService.ts` runs this inside a bare async IIFE,
*outside* the `wrapWithErrorHandling` wrappers used for keybindings/settings, so
the throw is unhandled and surfaces during app initialization.

## Why it matters

In production this is one of the highest-volume unhandled frontend errors —
~2.6k events across **~1,160 distinct sessions/day**, all funneling through this
one `Object.entries` call. Guarding the choke point silences it for every
caller.

## Fix

- Keep `registerCustomWidgets` typed `Record<string, ComfyWidgetConstructor>`
  (the correct internal contract) and early-return on nullish input. The runtime
  guard defends against untrusted extensions that violate the type at the
  boundary, without weakening the signature for legitimate callers.
- Add a regression test asserting `registerCustomWidgets(null!/undefined!)` does
  not throw (the `!` casts simulate the boundary violation).

## Test plan

- [x] `npx vitest run src/stores/widgetStore.test.ts` — 8 passing, including the
  new null/undefined case.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/stores/widgetStore.ts [hunk 1]
```diff
@@ -20,8 +20,13 @@ export const useWidgetStore = defineStore('widget', () => {
...
 
   function registerCustomWidgets(
-    newWidgets: Record<string, ComfyWidgetConstructor>
+    // Extensions are untrusted code: `getCustomWidgets` is typed to return
+    // `Record<string, ...>`, but in practice an extension can resolve it to
+    // null/undefined. Guard here so a single misbehaving custom node can't
+    // throw "Cannot convert undefined or null to object" and break app init.
+    newWidgets: Record<string, ComfyWidgetConstructor> | null | undefined
   ) {
+    if (!newWidgets) return
```
Comment (AustinMroz) [inline src/stores/widgetStore.ts:29]: I am strongly opposed to changing the type here. I do not think we should allow our API to be written by those who use it incorrectly.
```suggestion
    newWidgets: Record<string, ComfyWidgetConstructor>
  ) {
     // Guard for extensions incorrectly passing undefined
    if (!newWidgets) return
```

You can fix the tests with `null!` and `undefined!`

Comment (mattmillerai) [reply src/stores/widgetStore.ts:29]: Good call — reverted the type widening. Kept the runtime guard and switched the tests to `null!`/`undefined!` to simulate the contract violation. Confirmed it lints clean (we're on tseslint `recommended`, so no `no-unnecessary-condition` complaints about the guard).

One thing worth your read: the *most* boundary-faithful version of your principle would guard at the call site — `const widgets = await extension.getCustomWidgets(app); if (widgets) registerCustomWidgets(widgets)` — since that's where the untrusted value actually lands, and it drops the `!` casts entirely. I kept it centralized for the "fixed for every future caller" reason, but there's only one call site today. Happy to move it if you'd rather contain it at the edge.

**Review by AustinMroz — APPROVED** [review]
